tools/gotestloghelper/gotestevent/gotestevent_test.go

```diff
@@ -0,0 +1,349 @@
+package gotestevent
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/clihelper"
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/ptr"
+)
+
+type stdoutRedirector struct {
+	originalStdout *os.File    // To keep track of the original os.Stdout
+	r              *os.File    // Read end of the pipe
+	w              *os.File    // Write end of the pipe
+	outC           chan string // Channel to capture output
+	isClosed       bool
+}
+
+func newStdoutRedirector() *stdoutRedirector {
+	return &stdoutRedirector{isClosed: true}
+}
+
+func (sr *stdoutRedirector) startRedirect() {
+	// Save original stdout
+	sr.originalStdout = os.Stdout
+
+	// Create a pipe for capturing output
+	sr.r, sr.w, _ = os.Pipe()
+	os.Stdout = sr.w
+
+	// Initialize the output capture channel
+	sr.outC = make(chan string)
+
+	// Start capturing
+	go func() {
+		var buf io.ReadCloser = sr.r
+		b, _ := io.ReadAll(buf)
+		sr.outC <- string(b)
+	}()
+	sr.isClosed = false
+}
+
+func (sr *stdoutRedirector) closeRedirect() string {
+	if sr.isClosed {
+		return ""
+	}
+	// Close the write end of the pipe to finish the capture
+	sr.w.Close()
+
+	// Restore original stdout
+	os.Stdout = sr.originalStdout
+
+	// Read the captured output from the channel
+	out := <-sr.outC
+
+	sr.isClosed = true
+
+	return out
+}
+
+// do not use with parallel tests
```

Comment:
When should we use it?

Reply:
Just when I thought I had cleared out all the unused and commented out code there is a massive block of it here. Good catch